wherobots/db/connection.py

```diff
@@ -146,6 +144,8 @@ def __listen(self) -> None:
                 query.state = ExecutionState.COMPLETED
                 query.handler(self._handle_results(execution_id, results))
             elif query.state == ExecutionState.CANCELLED:
+                import pandas
```

Comment:
Why do this import lazily? I'd rather the module fail to load/import at the beginning, than introduce a subtle failure for the application developer on the first time they happen to cancel a query.